src/utils/helpers/translations/findUserRole.ts

```diff
@@ -0,0 +1,32 @@
+import { PermissionPolicyType } from '../../generated/graphql'
+import { UserRoles } from '../../types'
+
+/**
+ * @function: findUserRole
+ * Deduce the current user Role to interact with applications.
+ * Each UserRole define different view options and actions the user have access in the UI.
+ * - @param permissions - Array with group of permissions the user have for a template
+ * - @returns UserRole deduced from group of permissions or undefined.
+ */
+
+const userRoles: UserRoles = {
+  applicant: [PermissionPolicyType.Apply],
+  reviewer1: [PermissionPolicyType.Review],
+  reviewer2: [PermissionPolicyType.Review, PermissionPolicyType.Assign],
+  supervisor: [PermissionPolicyType.Assign],
+  consolidator: [PermissionPolicyType.Assign], //, PermissionPolicyType.Consolidate]
+}
+
+export default (permissions: Array<PermissionPolicyType>): string | undefined => {
+  const comparePermissions = permissions.map((permissionType) => permissionType.toUpperCase())
+
+  // Compare array of permission checking if are the same
+  const matching = Object.entries(userRoles).filter(([role, permissionList]) => {
+    console.log('role', role)
+
+    const difference = permissionList.filter((x) => !comparePermissions.includes(x))
+    return difference.length === 0
+  })
+  console.log('Possible user-roles:', Object.keys(matching))
+  return matching.length > 0 ? Object.keys(matching)[0] : undefined
+}
```

Comment:
This is the method that is returning `0` instead of `applicant`.

It seems overly confusing to me, I think it could be simpler. This seems to work (although probably needs some additional checks:
```
export default (permissions: Array<PermissionPolicyType>): string | undefined => {
  const comparePermissions = permissions.map((permissionType) => permissionType.toUpperCase())

  // Compare array of permission checking if are the same
  const matching = Object.entries(userRoles).filter(([role, permissionList]) => {
    const common = permissionList.filter((permission) => comparePermissions.includes(permission))
    return common.length > 0
  })
  return matching?.[0]?.[0] || undefined
```

Reply:
Ah, I see your other comment about fixing this elsewhere. Cool. I do think the method could be simpler, though, but I guess I'll take another look at it over on that PR.

Reply:
Done b5dc196